c7n/resources/cloudfront.py

```diff
@@ -181,9 +179,64 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('mismatch-s3-owner')
+class MismatchS3Owner(Filter):
+    """Check for existence of S3 bucket referenced by Cloudfront,
+       and verify whether owner is different from Cloudfront account owner.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: mismatch-s3-owner
+                resource: distribution
+                filters:
+                  - type: mismatch-s3-owner
+   """
+
+    schema = type_schema(
+        'mismatch-s3-owner')
+
+    permissions = ('s3:ListBuckets',)
+    retry = staticmethod(get_retry(('Throttling',)))
+
+    def is_s3_domain(self, domain_name):
+        if domain_name.endswith('.s3.amazonaws.com'):
+            return True
+        else:
+            return False
+
+    def get_s3_domain(self, domain_name):
+        if domain_name.endswith('.s3.amazonaws.com'):
+            return domain_name[:-len('.s3.amazonaws.com')]
+
+    def process(self, resources, event=None):
+        results = []
+
+        s3_client = local_session(self.manager.session_factory).client(
+            's3', region_name=self.manager.config.region)
+
+        buckets = {b['Name'] for b in s3_client.list_buckets()['Buckets']}
```

Comment:
no real reason to be using a dict in this case, should just a be a list in this case

Reply:
I was hoping to use a set here, as opposed to a list search, since it'll be more efficient from a performance perspective. Especially in cases where accounts have a large number of S3 buckets.

Reply:
Ah yeah... dicts hash their keys. Agree on set as there's no value to associate with the bucket